Audio to Dev merge

TimoV2003 opened the pull request:
Added the possibility to play .mp3 files, this currently has to be done in gameservice and will likely remain so for the foreseeable future

TimoV2003 committed:
Simple class with implemented library

TimoV2003 committed:
Merge branch 'Dev' into Audio

TimoV2003 committed:
Simple version of audio

Audio has been implemented in a different thread and the custom models have also been imported here

TimoV2003 committed:
Merge branch 'Dev' into Audio

TimoV2003 committed:
Music can now switch

TimoV2003 committed:
Merge branch 'Dev' into Audio

TimoV2003 committed:
Merge branch 'Dev' into Audio

Tiemenbr commented:
if you close the app it crashes, because the thread most likely doesn't close properly

TimoV2003 committed:
Fixed a few minor issues

Thread actually closes instead of crashing and fixed app crashing when trying to restart the game

TimoV2003 committed:
Merge branch 'Dev' into Audio

BOAD64 commented on CGR Template.vcxproj.filters:
waarom worden alle filters verwijderd?

BOAD64 commented on Source Files/GameService.cpp:
```diff
@@ -47,6 +49,8 @@ static bool showingDebugMenu = true;
 static double lastFrameTime = 0;
 static double deltaTime = 0.0f;
 
+bool switched = false;
```
onduildeijke naam. switched wat?

BOAD64 commented on Source Files/GameService.cpp:
```diff
@@ -77,6 +81,13 @@ void GameService::init() {
     floorFactory = std::make_shared<FloorFactory>();
 
+    if (switched == false)
+    {
```
{ hoord op same line te zijn niet nieuw line

BOAD64 commented on Source Files/GameService.cpp:
```diff
@@ -122,7 +133,14 @@ void GameService::init() {
 
-void GameService::update() {
+void GameService::update()
+{
```
{ hoord op same line te zijn niet nieuw line

BOAD64 commented on Source Files/GameService.cpp:
```diff
@@ -122,7 +133,14 @@ void GameService::init() {
+void GameService::update()
+{
+    if (!switched)
+    {
```
{ hoord op same line te zijn niet nieuw line

TimoV2003 committed:
Code Style Guide changes

TimoV2003 committed:
Merge branch 'Dev' into Audio

TimoV2003 committed:
Final Audio Commit
